src/main/scala/extracells/api/IStorageCellAdvanced.java

```diff
@@ -0,0 +1,105 @@
+package extracells.api;
+
+
+import appeng.api.storage.ICellWorkbenchItem;
+import appeng.api.storage.data.IAEItemStack;
+import net.minecraft.item.ItemStack;
+
+import javax.annotation.Nullable;
+
+
+/**
+ * Any item which implements this can be treated as an IMEInventory via
+ * Util.getCell / Util.isCell It automatically handles the internals and NBT
+ * data, which is both nice, and bad for you!
+ * <p>
+ * Good cause it means you don't have to do anything, bad because you have
+ * little to no control over it.
+ * <p>
+ * The standard AE implementation only provides 1-63 Types
+ */
+public interface IStorageCellAdvanced extends ICellWorkbenchItem
+{
+
+	/**
+	 * It wont work if the return is not a multiple of 8.
+	 * The limit is ({@link Integer#MAX_VALUE} + 1) / 8.
+	 *
+	 * @param cellItem item
+	 * @return number of bytes
+	 */
+	long getBytes( ItemStack cellItem );
+
+	/**
+	 * Determines the number of bytes used for any type included on the cell.
+	 *
+	 * @param cellItem item
+	 * @return number of bytes
+	 * @deprecated use {@link IStorageCell#getBytesPerType(ItemStack)}
+	 */
+	@Deprecated
+	int BytePerType( ItemStack cellItem );
+
+	/**
+	 * Determines the number of bytes used for any type included on the cell.
+	 *
+	 * @param cellItem item
+	 * @return number of bytes
+	 */
+	int getBytesPerType( ItemStack cellItem );
+
+	/**
+	 * Must be between 1 and 63, indicates how many types you want to store on
+	 * the item.
+	 *
+	 * @param cellItem item
+	 * @return number of types
+	 */
+	int getTotalTypes( ItemStack cellItem );
```

Comment:
Also if this interface is for huge single-type cells, why there assumptions are all over the code that types may be multiple?

Reply:
The constructor has no assumptions about the type limit and can be used for arbitrarily many types, the additions just happen to be single-type cells.

Reply:
Why didn't you just imported AE2 interface directly? It is really unlikely that our EC2 fork would be used w/o our AE2 fork...

Reply:
EC2 has a lot of copy/paste from AE2 because it had to allow being used with different AE2 versions, and could not rely on non-API classes

Reply:
It might just be me being bad with Java, but I didn't see a way to use the same interfaces as the primary functions that I need use Int instead of Long as the return type, hence the copies of the interfaces.

Reply:
Ah, original cells use int, weird, ok then.

Reply:
You still don't need deprecated method though, you don't have to maintain backwards compat